docs/advanced-guides/oauth-client.md

```diff
@@ -0,0 +1,283 @@
+
+The PAR request should have `Content-Type: application/x-www-form-urlencoded`.
+
+A successful response body will be a JSON object including the field `request_uri` (not to be confused with `redirect_uri`).
+
+### DPoP
+
+Clients must use DPoP to bind auth tokens to a specific client device or server. DPoP nonces, provided by the auth server, must be used.
+
+Clients generate a new DPoP cryptographic keypair for each auth session, and retain the keypair for the duration of the auth session. DPoP keypairs should never be exported or moved between devices, and should never be reused across users or between sessions for the same user. Client must start DPoP at the initial authorization request (PAR).
+
+`ES256` (NIST "P-256") is the cryptographic algorithm/curve which must be supported by all clients and auth servers. Browser Apps should use the WebCrypto API to generate non-exportable keypairs. Other clients may find implementations of this cryptographic system in generic JWT libraries, or in generic cryptographic libraries for their language or environment. DPoP is also increasingly required as part of OAuth profiles and will hopefully be support by generic OAuth libraries.
+
+DPoP involves setting a HTTP Header (`DPoP`) on every token request and every authorized request to the PDS. The header value is a self-signed JWT. There is random field in the body, and JWTs are generated and signed uniquely for every requests (DPoP JWTs should not be reused between requests).
+
+The server returns the current DPoP nonce in the `DPoP-Nonce` HTTP header in every response. Nonce values may be shared across all users and sessions on the server, or may be scoped to individual users and sessions. Nonces may be shared between access token use (PDS requests) and authorization server requests (PDS or entryway), but they may be distinct servers, so clients should always track DPoP nonces separately for the two uses. Nonces change periodically, with a rotation period chosen by the server. Clients should persist the DPoP nonce for each session, and update the persisted value when a response is received with a different value. If the nonce is missing, or has become outdated, the server will return an HTTP 400 response with JSON body and the `error` field set to `use_dpop_nonce`. The response will include the correct nonce in the `DPoP-Nonce` header, and the client can retry the request. The client discovers the initial nonce for each server by doing this request/error/retry cycle at least once. Servers will usually accept stale/old nonce values for a short time window to reduce errors-and-retries caused by clients making multiple concurrent authorized requests. Ideally the request/error/retry cycle does not need to happen again, though clients should be ready for it at any time (eg, if the nonce has rotated multiple times between requests).
```

Comment:
For detecting the `use_dpop_nonce` error, one should use the `WWW-Authenticate` HTTP header, when communicating with the resource server:

https://datatracker.ietf.org/doc/html/rfc9449#section-9

```
 HTTP/1.1 401 Unauthorized
 WWW-Authenticate: DPoP error="use_dpop_nonce", \
   error_description="Resource server requires nonce in DPoP proof"
 DPoP-Nonce: eyJ7S_zG.eyJH0-Z.HX4w-7v
```

Reply:
During PAR, the error mechanism, defined by OAuth, is indeed to return an `error` field in the JSON payload.